app/controllers/CancelDeclarationController.scala

```diff
@@ -54,13 +63,31 @@ class CancelDeclarationController @Inject()(
         (formWithErrors: Form[CancelDeclaration]) =>
           Future.successful(BadRequest(cancelDeclarationPage(formWithErrors))),
         form => {
+          auditService.auditAllPagesUserInput(Json.toJson(form).as[JsObject])
           val context = exportsMetrics.startTimer(cancelMetric)
-          customsDeclareExportsConnector.createCancellation(form).map { _ =>
-            exportsMetrics.incrementCounter(cancelMetric)
-            context.stop()
-            Ok(cancelConfirmationPage())
-          }
+          for {
+            _ <- customsDeclareExportsConnector.createCancellation(form) andThen {
```

Comment:
Are we able to move this from for-comprehension ?

Reply:
It would looks like
```
customsDeclareExportsConnector.createCancellation(form).andThen {
  case Failure(exception) =>
    logger.error(s"Error response from backend $exception")
    auditService.audit(AuditTypes.Cancellation, auditData(form, Failure.toString))
  casee Success(_) => 
    auditService.audit(AuditTypes.Cancellation, auditData(form, Success.toString))
    exportsMetrics.incrementCounter(cancelMetric)
    context.stop()
}.map(_ => Ok(cancelConfirmationPage())
```

Reply:
This has been addressed.